app/assets/javascripts/validation.js

```diff
@@ -10,69 +10,228 @@ $(document).ready(function () {
+    styleElement(element, 'form-element-valid', 'form-element-invalid');
+    renderErrorMsg(element, errorMsg);
+  }
+
+  this.isFormValid();
+};
+
+SignUpFormValidator.prototype.disableSubmitBtn = function() {
+  this.submitBtn.setAttribute('disabled', '');
+};
+
+SignUpFormValidator.prototype.enableSubmitBtn = function() {
+  this.submitBtn.removeAttribute('disabled');
+};
+
+SignUpFormValidator.prototype.isFormValid = function() {
+  // Form is valid if all elements have passsed validation successfully
+  var isValid =
+    Object.values(this.validationTracker).filter(Boolean).length === 4;
+
+  if (isValid) this.enableSubmitBtn();
+  else this.disableSubmitBtn();
+};
+
+function validateUsername(obj) {
+  var username = this.value;
+  var self = this;
+
+  if (username.length < 3) {
+    restoreOriginalStyle(this);
+  } else {
+    $.get('/api/srch/profiles?query=' + username, function(data) {
+      if (data.items) {
+        $.map(data.items, function(userData) {
+          if (userData.doc_title === username) {
+            obj.updateUI(self, false, 'Username already exists');
+          } else {
+            obj.updateUI(self, true);
+          }
+        });
+      } else {
+        obj.updateUI(self, true);
       }
-    },
-    submitHandler: function(form) {
-      form.submit();
-    }
-  });
+    });
+  }
+}
+
+function validateEmail(obj) {
+  var email = this.value;
+  var regexp = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  var isValid = regexp.test(email);
+
+  obj.updateUI(this, isValid, 'Invalid email');
+}
+
+function validatePassword(confirmPasswordElement, obj) {
+  var password = this.value;
+
+  if (!isPasswordValid(password)) {
+    obj.updateUI(
+      this,
+      false,
+      'Please make sure password is atleast 8 characters long with minimum one numeric value.'
+    );
+    return;
+  }
+
+  if (password === confirmPasswordElement.value) {
+    obj.updateUI(confirmPasswordElement, true);
+  }
+
+  obj.updateUI(this, true);
+}
+
+function validateConfirmPassword(passwordElement, obj) {
+  var confirmPassword = this.value;
+  var password = passwordElement.value;
+
+  if (confirmPassword !== password) {
+    obj.updateUI(this, false, 'Passwords must be equal');
+    return;
+  }
+
+  obj.updateUI(this, true);
+}
+
+// Password is valid if it is at least 8 characaters long and contains a number
+// This is password's validation logic, no UI
+function isPasswordValid(password) {
+  var doesContainNum = /\d+/g.test(password);
+  var isValid = password.length >= 8 && doesContainNum;
+
+  return isValid;
+}
+
+function renderErrorMsg(element, message) {
+  if (!message) return;
+
+  // Error messages are rendered inside of a <small> HTML element
+  var errorMsgElement = element.nextElementSibling;
+  if (!errorMsgElement) {
+    // On pl.org/register invalid elements are wrapped in a div.
```

Comment:
Pl-> publiclab
Please don't use short forms while writing documentation. Documentation is for new developer

Reply:
:+1: